packages/victory-core/src/victory-state/victory-provider-maybe.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { useVictoryContextMaybe, VictoryProvider } from "./victory-provider";
+// TODO: Fix this dependency:
+import { VictoryCommonProps, Clone } from "../victory-util";
+
+type ContainerProp = Required<Pick<VictoryCommonProps, "containerComponent">>;
+
+/* eslint-disable react/prop-types */
+
+export function withVictoryProvider<
+  TComp extends React.FC<TProps>,
+  TProps extends React.PropsWithChildren<ContainerProp>,
+>(Comp: TComp): TComp {
+  const WithProvider = React.memo((props: TProps) => {
+    const updateChildProps = useVictoryContextMaybe(
+      (value) => value?.updateChildProps,
+    );
+    const hasParentProvider = !!updateChildProps;
+
+    React.useEffect(() => {
+      if (!hasParentProvider) return;
+
+      const id = Symbol("WithProvider");
+      updateChildProps(id, props);
```

Comment:
Is this id just a placeholder? I'm guessing we will need some way of providing unique IDs for each child.

Reply:
FWIW, Grant and I have discussed some of the flaws with this `updateChildProps` approach -- esp. SSR support.
We have some ideas how to achieve a better flow, which will make this code obsolete in a future PR.